modules/delegates.js

```diff
@@ -589,16 +597,33 @@ Delegates.prototype.isLoaded = function () {
 Delegates.prototype.internal = {
 	forgingEnable: function (req, cb) {
 		library.schema.validate(req.body, schema.enableForging, function (err) {
+
 			if (err) {
 				return setImmediate(cb, err[0].message);
 			}
 
-			var keypair = library.ed.makeKeypair(crypto.createHash('sha256').update(req.body.secret, 'utf8').digest());
+			var keypair;
+			var encryptedList, decryptedSecret, encryptedItem;
+			encryptedList = library.config.forging.secret;
 
-			if (req.body.publicKey) {
-				if (keypair.publicKey.toString('hex') !== req.body.publicKey) {
+			encryptedItem = _.find(encryptedList, function (item) {
+				return item.publicKey === req.body.publicKey;
+			});
+
+			if (!!encryptedItem) {
+				try {
+					decryptedSecret = __private.decryptSecret(encryptedItem.encryptedSecret, req.body.key);
+				} catch (e) {
 					return setImmediate(cb, 'Invalid passphrase');
 				}
+	
+				keypair = library.ed.makeKeypair(crypto.createHash('sha256').update(decryptedSecret, 'utf8').digest());
+			} else {
+				return setImmediate(cb, ['Delegate with publicKey:', req.body.publicKey, 'not found'].join(' '));
+			}
+
+			if (keypair.publicKey.toString('hex') !== req.body.publicKey) {
+				return setImmediate(cb, 'Invalid key and public key combination');
```

Comment:
Can you put all of this logic into a private method since its all the same logic as disable? Will make this much clearer/smaller.